api/server.go

```diff
@@ -16,24 +17,49 @@ import (
 	"github.com/algorand/indexer/idb"
 )
 
+// ExtraOptions are options which change the behavior or the HTTP server.
+type ExtraOptions struct {
+	// Tokens are the access tokens which can access the API.
+	Tokens []string
+
+	// DeveloperMode turns on features like AddressSearchRoundRewind
+	DeveloperMode bool
+
+	// MetricsEndpoint turns on the /metrics endpoint for prometheus metrics.
+	MetricsEndpoint bool
+
+	// MetricsEndpointVerbose generates separate histograms based on query parameters on the /metrics endpoint.
+	MetricsEndpointVerbose bool
+}
+
 // Serve starts an http server for the indexer API. This call blocks.
-func Serve(ctx context.Context, serveAddr string, db idb.IndexerDb, fetcherError error, log *log.Logger, tokens []string, developerMode bool) {
+func Serve(ctx context.Context, serveAddr string, db idb.IndexerDb, fetcherError error, log *log.Logger, options ExtraOptions) {
 	e := echo.New()
 	e.HideBanner = true
 
+	if options.MetricsEndpoint {
+		p := prometheus.NewPrometheus("indexer", nil, nil)
```

Comment:
I'm guessing from context that labstack/echo-contrib/prometheus is providing the handler for `/metrics` and it will provide per-URL-path counts, like `/metrics: 123`, but it would be nice to see a comment describing what's going on. Also I don't see where any handler is specifically bound to `/metrics`. Where does that happen?

Reply:
That all happens in `p.Use(e)`, which is weird since middleware is normally registered with `e.Use(p)`. I guess the reason for this is because `p.Use` calls `e.Use` for stat collection and also registers the `/metrics` handler.

I added some comments to the code.